client/lib/query-args/test/index.js

```diff
@@ -0,0 +1,74 @@
+import page from 'page';
+import useQueryArgs from '../index';
+
+jest.mock( 'react' );
+jest.mock( 'page' );
+
+describe( '#useQueryArgs', () => {
+	const queryArgs = useQueryArgs();
+
+	global.window = {
+		location: {
+			href: '',
+		},
+	};
+
+	test( 'should remove the query if search is empty', () => {
+		global.window.location.pathname = '/read/search';
+		global.window.location.href = 'https://wordpress.com/read/search?q=test';
+		const expectedResult = '/read/search';
+		queryArgs.setQueryArgs();
```

Comment:
Could we force people to be specific here?

```suggestion
		queryArgs.setQueryArgs( {} );
```

And on that note, should `setQueryArgs( {} )` be clearing existing query params? (implementation above doesn't)

I kind of think it should be but we might want an `updateQueryArgs( {} )` which only updates what you give it leaving other params in place.

The original implementation only updates `?s=term` but leaves any other values in place so I'm guessing there might be a need to allow for that but I think most of the time people will just want setQueryArgs.

Reply:
Sounds good @lsl, add the `updateQueryArgs` [in this commit](https://github.com/Automattic/wp-calypso/pull/63825/commits/ee0eeac760cda69b648a65010ee01024b524042b). Now, `setQueryArgs( {} )` will clear all parameters.